Lib/selectors.py

```diff
@@ -580,16 +581,31 @@ def close(self):
             super().close()
 
 
+def _can_use(method):
+    """Check if we can use the selector depending upon the
+    operating system. """
+    # Implementation based upon https://github.com/sethmlarson/selectors2/blob/master/selectors2.py
+    try:
+        if method == 'poll':
+            poll_obj = select.poll()
+            poll_obj.poll(0)
+        else:
+            getattr(select, method)().close()
+        return True
+    except (OSError, AttributeError):
+        return False
```

Comment:
I suggest to rewrite the ``_can_use`` function to make it a bit easier to understand and to debug:

```suggestion
    selector = getattr(select, method, None)
    if selector is None:
         # select module does not implement method
         return False
    # check if the OS and Kernel actually support the method. Call may fail with
    # OSError: [Errno 38] Function not implemented
    try:
        selector_obj = selector()
        if method == 'poll':
            # check that poll actually works
            selector_obj.poll(0)
        else:
            # close epoll, kqueue, and devpoll fd
            selector_obj.close()
        return True
    except OSError:
        return False
```

Reply:
+1, I think this approach is significantly more readable. 

But I see that it still has the added guard for missing `select`. @tiran did you change your mind about this from the previous comment or find a platform that we support which doesn't have select?

Reply:
@aeros I changed the extra `select` check.